five-minute-serverless-backend/README.md

```diff
@@ -0,0 +1,209 @@
+      },
+    };
+    if (context.request.method !== 'OPTIONS') {
+      const output = await handler(body);
+      response.headers['Content-Type'] = 'application/json';
+      response.body = JSON.stringify(output);
+    }
+    return new context.HTTPResponse(response);
+  };
+;
+
+exports.handler = CORS(async (body, context) => {
+  return (typeof(body) === 'string' ? body : '').toUpperCase();
+});
+```
+
+## Call your function from frontend code
+
+Our backend is now live and we can use a simple HTML app to call it. The app code is in a single file named `frontend.html`.
+
+First you need to plug in your Binaris account ID. At the very top of the file, locate the following line:
+
+```htmlmixed
+    <script>const BINARIS_ACCOUNT_ID = '<account-id>'</script>
+```
+
+Replace `<account-id>` with your actual account ID.
+
+You can now open `frontend.html` in your browser to run the app. Simply type in some text and hit Enter. The text will be sent to the backend and returned in upper-case to be displayed on screen.
+
+Calling our backend is a simple HTTP POST request. We included our own wrapper for `XMLHttpRequest` mostly to handle HTTP request and response headers. Our wrapper, called `request()` also returns a Promise so that the code calling it is cleaner.
+
+We then implemented a helper function to invoke our backend cloud function:
+
+```javascript
+const invoke = async (name, input) => {
+  const url = `https://run.binaris.com/v2/run/${BINARIS_ACCOUNT_ID}/${name}`;
+  const data = JSON.stringify(input);
+  const headers = { 'Content-Type': 'application/json' };
+  const res = await sendHttpRequest('post', url, data, headers);
+  const ct = res.headers['content-type'];
+  if (res.status !== 200 || !ct || !ct.startsWith('application/json')) {
+    throw new Error(`Backend error: status=${res.status} response: ${res.body}`);
+  }
+  return JSON.parse(res.body);
+};
+```
+
+This helper function uses the Binaris URL structure and the account ID configured above to form the backend URL. It also handles JSON encoding and decoding, and sets and verifies content type headers.
+
+All that's left now is to call our backend whever the user hits Enter:
+
+```javascript
+$input.addEventListener('keyup', async e => {
+  if (e.keyCode === 13 && $input.value) {
+    const value = await invoke('public_backend', $input.value);
+    $output.innerHTML += value + '\n';
+    $input.value = '';
+  }
+});
+```
+
+The actual invocation is a simple one-liner calling our helper function with the name of our serverless function and the input object. In case of error, invoke throughs an exception and the on-screen status remains unchanged.
+
+## What's next?
+
+That's it! We've seen how simple and fast it is to create a scalable cloud backend using Binaris serverless functions. Visit our [developer site](https://dev.binaris.com/) for more tutorials and documentation. Vist our [binaris.com](https://www.binaris.com/) to learn more about the platform.
```

Comment:
I suspect the final sentence should be `Visit out [homepage]`

Reply:
Actually "Visit our" and not "Vist our" or "Visit out" :-) Fixed